src/core/etl/tests/Flow/ETL/Tests/Unit/Transformer/EntryExpressionEvalTransformerTest.php

```diff
@@ -94,7 +94,7 @@ public function test_xml_xpath_expression_when_there_is_more_than_one_node_under
         $xpath = new \DOMXPath($document);
 
         $this->assertEquals(
-            Entry::list_of_objects('xpath', \DOMNode::class, [
+            Entry::list_of_objects('xpath', \DOMElement::class, [
```

Comment:
hmm why?

Reply:
Cause: `$xpath->query('/root/foo')->item(0)` returns `DOMElement` not just `DOMNode`:
```console
object(DOMElement)#409 (23) {
  ["schemaTypeInfo"]=>
  NULL
  ["tagName"]=>
  string(3) "foo"
  ["firstElementChild"]=>
  NULL
  ["lastElementChild"]=>
  NULL
  ["childElementCount"]=>
  int(0)
  ["previousElementSibling"]=>
  NULL
  ["nextElementSibling"]=>
  string(22) "(object value omitted)"
  ["nodeName"]=>
  string(3) "foo"
  ["nodeValue"]=>
  string(3) "bar"
  ["nodeType"]=>
  int(1)
  ["parentNode"]=>
  string(22) "(object value omitted)"
  ["childNodes"]=>
  string(22) "(object value omitted)"
  ["firstChild"]=>
  string(22) "(object value omitted)"
  ["lastChild"]=>
  string(22) "(object value omitted)"
  ["previousSibling"]=>
  NULL
  ["nextSibling"]=>
  string(22) "(object value omitted)"
  ["attributes"]=>
  string(22) "(object value omitted)"
  ["ownerDocument"]=>
  string(22) "(object value omitted)"
  ["namespaceURI"]=>
  NULL
  ["prefix"]=>
  string(0) ""
  ["localName"]=>
  string(3) "foo"
  ["baseURI"]=>
  string(29) "/Users/stloyd/Documents/flow/"
  ["textContent"]=>
  string(3) "bar"
}

Reply:
so how it was working before?

Reply:
Before there was a hack:
```php
            if ($class !== null) {
                if ($class === \DateTimeImmutable::class || $class === \DateTime::class) {
                    $class = \DateTimeInterface::class;
                }

                if ($class === \DOMElement::class) {
                    $class = \DOMNode::class;
                }

                return new Entry\ListEntry($entryName, ListElement::object($class), $value);
            }
```
Now it stores the class that was detected on an object.